qcodes/dataset/data_export.py

```diff
@@ -347,24 +369,26 @@ def datatype_from_setpoints_2d(xpoints: np.ndarray,
     return 'unknown'
 
 
-def reshape_2D_data(x: np.ndarray, y: np.ndarray,
-                    z: np.ndarray) -> Tuple[np.ndarray, np.ndarray,
-                                            np.ndarray]:
+def reshape_2D_data(x: np.ndarray, y: np.ndarray, z: np.ndarray
+                    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
     xrow = np.array(_rows_from_datapoints(x)[0])
     yrow = np.array(_rows_from_datapoints(y)[0])
     nx = len(xrow)
     ny = len(yrow)
 
-
     # potentially slow method of filling in the data, should be optimised
     log.debug('Sorting 2D data onto grid')
-    z_to_plot = np.full((ny, nx), np.nan)
+
+    if isinstance(z[0], str):
+        z_to_plot = np.full((ny, nx), ' '*int(str(z.dtype)[2:]))
```

Comment:
Please, add some comments to the `str(z.dtype)[2:]` part. Smth like:
```suggestion
        # strings in numpy have dtype names that look like '<U25';
        # `str(z.dtype)[2:]` extracts that "dtype length" (25 from above),
        # this is needed in order to allocate space for strings in numpy array:
        # once the strings will be assigned, the dtype of specific element will
        # automatically adjust to the string length in that array element
        z_to_plot = np.full((ny, nx), ' '*int(str(z.dtype)[2:]))
```